File: Server/log-nine-backend/Models/Team.cs
using System.ComponentModel.DataAnnotations;

namespace LogNineBackend.Models;

public class Team {
    public int Id { get; set; }
    
    [Required]
    [StringLength(100, MinimumLength = 1)]
    public string Name { get; set; } = string.Empty;
    
    public int BoardId { get; set; }
    public Board Board { get; set; } = null!;

    [Range(0, 9999)]
    public float SrFrequency { get; set; }
    
    [Range(0, 9999)]
    public float LrFrequency { get; set; }
}

public record struct TeamDTO(int Id, string Name, int BoardId, float SrFrequency, float LrFrequency); 

File: Webpage/src/entities/Team.ts
type Team = {
  id: number;
  boardId: number;
  name: string;
  srFrequency: number;
  lrFrequency: number;
};

export const UndefinedTeam: Team = {
  id: -1,
  boardId: -1,
  name: "Undefined",
  srFrequency: -1,
  lrFrequency: -1,
};

export default Team;
